types/types.go

```diff
@@ -87,6 +88,16 @@ const (
 	ResourceLike = Type(rune(byteResource))
 	// FunctionLike is the underlying type of all functions
 	FunctionLike = Type(rune(byteFunction))
+
+	// StringSlice is used to represent special function for searching strings.
+	// Users are never exposed to this type directly and it is not documented
+	// as a primitive. It serves as a way to array functions on top of strings,
```

Comment:
Not sure this next sentence makes sense "serves as a way to array functions  on top of strings"